es-vue-base/src/lib-components/EsFileUpload.vue

```diff
@@ -0,0 +1,254 @@
+            // selected the files from the file picker which limits the file types to the ones specified in the
+            // fileTypes prop.
+            this.active = false;
+            this.files = [];
+
+            // Use DataTransferItemList interface to access the file(s)
+            const invalidFiles = [...event.dataTransfer.items].filter((item) => item.kind !== 'file');
+
+            const dataTransfersAsFiles = [...event.dataTransfer.items]
+                .filter((item) => item.kind === 'file')
+                .map((item) => item.getAsFile());
+
+            const validFiles = dataTransfersAsFiles
+                .filter((file) => this.fileTypes.includes(file.type));
+            const fileErrors = dataTransfersAsFiles
+                .filter((file) => !this.fileTypes.includes(file.type))
+                .concat(invalidFiles)
+                .map((item) => item.name);
+            if (fileErrors.length > 0) {
+                this.$emit('fileTypeError', fileErrors);
+            }
+
+            this.files = validFiles;
+            if (this.files.length > 0 && this.fileSizeValid()) {
+                this.$emit('readyToUpload', this.files.length);
+                this.readFilesIntoUrl(validFiles);
+            }
+        },
+        openFilePicker() { document.getElementById('fileInput').click(); },
+        upload() {
+            this.files.forEach((file, index) => {
+                this.uploadSingleFile(file, index);
+            });
+        },
+        uploadSingleFile(file, index) {
+            const config = {
+                headers: {
+                    'Content-Type': file.type,
+                },
+                onUploadProgress: (progressEvent) => {
+                    const percentCompleted = Math.round((progressEvent.loaded * 100) / progressEvent.total);
+                    this.$emit('uploadProgress', { fileName: file.name, percentCompleted });
+                    return percentCompleted;
+                },
+            };
+            this.$axios.post(
+                this.uploadUrls[index],
+                file,
+                config,
+            )
+                .then((response) => {
+                    if (response.status === this.uploadSuccessStatusCode) {
+                        this.$emit('uploadSuccess', file.name);
+                    } else {
+                        this.$emit('uploadFailure', file.name);
+                    }
+                })
+                .catch(() => {
+                    this.$emit('uploadFailure', file.name);
+                });
+        },
+    },
+};
+</script>
+<style lang="scss" scoped>
+@import '~@energysage/es-bs-base/scss/bootstrap';
+@import "~@energysage/es-bs-base/scss/variables";
+
+.es-file-upload {
+    align-items: center;
+    background-color: $gray-200;
+    border-radius: $border-radius;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
```

Comment:
A lot of these can be done without custom css using BS4 class names.

`align-items-center bg-gray-200 rounded d-flex flex-column justify-content-center`